src/interpreter.ts

```diff
@@ -1,21 +1,62 @@
 
-import { Graph } from 'graphir';
+import { Graph, Value, NonTerminalControlVertex, ControlVertex } from 'graphir';
+import { State } from './state'
+import * as evaluate from './evaluate'
 
-
-export class ExecutionResult{
-    constructor() {}
+export class ExecutionResult {
+    
+    private _finalState: State;
+    private _returnedValue: Value;
+    
+    constructor(finalState: State, returnedValue: Value) {
+        this._finalState = finalState;
+        this._returnedValue = returnedValue;
+    }
 
     queryVertexValue(id: number): unknown {
-        //TODO: implement
-        return undefined;
-    };
-
+        return this._finalState.getVertexValue(id);
+    }
 
     returnValue(): unknown {
-        return 1;
+        return this._returnedValue;
     }
 }
 
 export function executeGraph(graph: Graph): ExecutionResult {
-    return new ExecutionResult();
+    if (graph.verify() == false) {
+        throw new Error(`Invalid graph input`);
+    }
+    let graphInterp = GraphInterpreter.getInstance();
+    let nodeValue: Value = 0;
+    let currentNode: NonTerminalControlVertex = graph.startVertex!.next!;
+
+    while (currentNode !== undefined) {
+        if (currentNode.verify() == false) {
+            throw new Error(`Invalid vertex`);
+        }
+        if (graphInterp.state.vertexValueExists(currentNode.id) == false) {
+            nodeValue = evaluate.evaluateControlNode(currentNode);
```

Comment:
Wont this fail the execution as soon as we will have more control vertices in the input graph?
For example, if we have a graph that looks like this: Start -> Pass -> Return, an error will be thrown for the second vertex.